### PR TITLE
Closes #2656 `stats_test.py` conversion for new test framework

### LINKED ISSUE #2656: `stats_test.py` Conversion for new test framework

### PR DESCRIPTION
**Pull request by TheOddCzar789**
This PR (Closes https://github.com/Bears-R-Us/arkouda/issues/2656):

Converts file to new test framework.

Adds looping/test grouping for brevity.

### CODE REVIEW

File: PROTO_tests/tests/stats_test.py [hunk 1]
```diff
@@ -0,0 +1,179 @@
+import numpy as np
+import pandas as pd
+import pytest
+
+import arkouda as ak
+
+
+class TestStats:
+    def setup_class(cls):
```
Comment (pierce314159) [inline PROTO_tests/tests/stats_test.py:9]: this should have an `@classmethod` decorator

File: PROTO_tests/tests/stats_test.py [hunk 2]
```diff
@@ -0,0 +1,179 @@
...
+            assert self.x.cov(ark) == pytest.approx(self.pdx.cov(pand))
+            assert self.b.cov(ark) == pytest.approx(self.pdb.cov(pand))
+            assert self.f.cov(ark) == pytest.approx(self.pdf.cov(pand))
+            assert self.x.corr(ark) == pytest.approx(self.pdx.corr(pand))
+            assert self.b.corr(ark) == pytest.approx(self.pdb.corr(pand))
+            assert self.f.corr(ark) == pytest.approx(self.pdf.corr(pand))
+
+        # test uint with self (other cases covered above)
+        assert self.u.cov(self.u) == pytest.approx(self.pdu.cov(self.pdu))
+        assert self.u.corr(self.u) == pytest.approx(self.pdu.corr(self.pdu))
```
Comment (pierce314159) [inline PROTO_tests/tests/stats_test.py:75]: you could use `getattr` to cut down on the lines of code here
```python
for fn in "cov", "corr":
    for arg in [
        getattr(self.pdx, fn)(self.pdy),
        getattr(self.y, fn)(self.x),
        getattr(ak, fn)(self.x, self.y),
        getattr(ak, fn)(self.y, self.x),
    ]:
        assert getattr(self.x, fn)(self.y) == pytest.approx(arg)

    # test int, bool, and float with other types
    for ark, pand in [
        (self.u, self.pdu),
        (self.b, self.pdb),
        (self.f, self.pdf),
    ]:
        assert getattr(self.x, fn)(ark) == pytest.approx(getattr(self.pdx, fn)(pand))
        assert getattr(self.b, fn)(ark) == pytest.approx(getattr(self.pdb, fn)(pand))
        assert getattr(self.f, fn)(ark) == pytest.approx(getattr(self.pdf, fn)(pand))

    # test uint with self (other cases covered above)
    assert getattr(self.u, fn)(self.u) == pytest.approx(getattr(self.pdu, fn)(self.pdu))
```

Comment (TheOddCzar789) [reply PROTO_tests/tests/stats_test.py]: Updated

File: PROTO_tests/tests/stats_test.py (hunk 1 above)
Comment (TheOddCzar789) [reply PROTO_tests/tests/stats_test.py:10]: Added

**Review by jaketrookman — APPROVED** [review]
Looks good

File: PROTO_tests/tests/stats_test.py [hunk 3]
```diff
@@ -0,0 +1,168 @@
...
+                (self.u, self.pdu),
+                (self.b, self.pdb),
+                (self.f, self.pdf),
+            ]:
+                assert getattr(self.x, fn)(ark) == pytest.approx(getattr(self.pdx, fn)(pand))
+                assert getattr(self.b, fn)(ark) == pytest.approx(getattr(self.pdb, fn)(pand))
+                assert getattr(self.f, fn)(ark) == pytest.approx(getattr(self.pdf, fn)(pand))
+
+            # test uint with self (other cases covered above)
+            assert getattr(self.u, fn)(self.u) == pytest.approx(getattr(self.pdu, fn)(self.pdu))
```
Comment (pierce314159) [inline PROTO_tests/tests/stats_test.py:64]: Now that we have this nice condensed format, I don't see a reason to special case `u`
```suggestion
                assert getattr(self.f, fn)(ark) == pytest.approx(getattr(self.pdf, fn)(pand))
                assert getattr(self.u, fn)(ark) == pytest.approx(getattr(self.pdu, fn)(pand))
```

**Review by pierce314159 — APPROVED** [review]
One suggestion but looks good otherwise

**Review by jaketrookman — APPROVED** [review]

**Review by pierce314159 — APPROVED** [review]

**Pull request merged**